src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4711,43 +4712,7 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers, bool isFunctio
                     continue;
                 }
 
-                var modifier = this.EatToken();
```

Comment:
syntactic diagnostics break incrementality and should be moved to binder, unless it would cause a 'smearing peanut butter' effect.